features/netsocket/nsapi_ppp.h

```diff
@@ -0,0 +1,34 @@
+/** \addtogroup netsocket */
+/** @{*/
+/* nsapi_ppp.h
+ * Modified work Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef NSAPI_PPP_H_
+#define NSAPI_PPP_H_
+
+#include "FileHandle.h"
+#include "NetworkStack.h"
+
+namespace mbed {
+
+NetworkStack *nsapi_ppp_get_stack();
```

Comment:
For pbuf implementation will be needed (compiled in) only if you use EMAC APIs, so it's different. So is this PR gated by your implementation of thisAPI for nanostack? I think we should revisit that as part of https://github.com/ARMmbed/mbed-os/pull/4079 and add flags `DEVICE_ETH` `DEVICE_WIFI` `DEVICE_PPP` rather than DEVICE_LWIP, device should claim support for technologies/features and mbed should decide what IP stack is being used. It would also guard us against things like that, and only enabling PPP would make these APIs being compiled in and it would also pull LWIP in.